src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp

```diff
@@ -284,18 +299,197 @@ ZeroInferRequest::ZeroInferRequest(const std::shared_ptr<ZeroInitStructsHolder>&
+    if (res == ZE_RESULT_SUCCESS) {
+#ifdef _WIN32
+        if (desc.id) {
+#endif
+            switch (desc.type) {
+            case ZE_MEMORY_TYPE_HOST:
+            case ZE_MEMORY_TYPE_DEVICE:
+            case ZE_MEMORY_TYPE_SHARED:
+                _copyAllTensors[name] = tensor;
+                setTensorData = true;
+                break;
+            case ZE_MEMORY_TYPE_UNKNOWN:
+            case ZE_MEMORY_TYPE_FORCE_UINT32:
+                break;
+            }
+#ifdef _WIN32
+        }
+#endif
+    }
+
+    if (!setTensorData) {
+        if (_copyAllTensors.find(name) != _copyAllTensors.end()) {
+            auto& copyTensor = _copyAllTensors.at(name);
+            auto remoteCopyTensor = std::dynamic_pointer_cast<ZeroRemoteTensor>(copyTensor);
+
+            if (remoteCopyTensor != nullptr) {
+                ov::Allocator allocator;
+                if (isInput && (_properties.flags & ZE_DEVICE_PROPERTY_FLAG_INTEGRATED)) {
+                    allocator = zeroMemory::HostMemAllocator(_initStructs, ZE_HOST_MEM_ALLOC_FLAG_BIAS_WRITE_COMBINED);
+                } else {
+                    allocator = zeroMemory::HostMemAllocator(_initStructs);
+                };
+
+                _copyAllTensors.at(name) = ov::make_tensor(tensor->get_element_type(), tensor->get_shape(), allocator);
+                setTensorData = true;
+            }
+        }
+    }
+
+    if (setTensorData) {
+        _tensorsData[name] = TensorData{_copyAllTensors.at(name)->data(),
+                                        _copyAllTensors.at(name)->get_byte_size(),
+                                        _createPipeline ? false : true};
+        _tensorIsDifferent = true;
+    }
+}
+
+void ZeroInferRequest::get_remote_tensor_mem(std::shared_ptr<ZeroRemoteTensor> tensor, std::string name) {
+    auto l0_context = reinterpret_cast<ze_context_handle_t>(
+        extract_object(tensor->get_context()->get_property(), ov::intel_npu::l0_context));
+    if (_initStructs->getContext() != l0_context) {
+        OPENVINO_THROW("Using different context for creating the tensor is not supported");
+    }
+
+    auto data = extract_object(tensor->get_properties(), ov::intel_npu::mem_handle);
+    if (data == nullptr) {
+        OPENVINO_THROW("Empty buffer");
+    }
+
+    _copyAllTensors[name] = tensor;
+    _tensorsData[name] = TensorData{data, tensor->get_byte_size(), _createPipeline ? false : true};
+    _tensorIsDifferent = true;
+}
+
+void ZeroInferRequest::check_and_get_mem_data(std::string name, bool isInput) {
+    auto& tensor = _allTensors.at(name);
+    auto remoteTensor = std::dynamic_pointer_cast<ZeroRemoteTensor>(tensor);
+
+    if (remoteTensor == nullptr) {
+        _logger.debug("ZeroInferRequest::check_and_get_mem_data - set new tensor");
+        get_tensor_data(tensor, name, isInput);
+    } else {
+        _logger.debug("ZeroInferRequest::check_and_get_mem_data - set new remote tensor");
+        get_remote_tensor_mem(remoteTensor, name);
```

Comment:
I find it a bit odd one is called `mem` and the other `data`. I believe both are true in both contexts, so there is no need for a divergence in naming.

Reply:
I used data for data returned by ze functions and mem for the mem_handle received from the user. Please let me know if you have a better idea for naming them.